java-frontend/src/main/java/org/sonar/java/JavaFrontend.java

```diff
@@ -38,19 +38,21 @@
 import org.sonar.java.ast.visitors.FileLinesVisitor;
 import org.sonar.java.ast.visitors.SyntaxHighlighterVisitor;
 import org.sonar.java.collections.CollectionUtils;
-import org.sonarsource.analyzer.commons.collections.ListUtils;
 import org.sonar.java.filters.SonarJavaIssueFilter;
 import org.sonar.java.model.JParserConfig;
 import org.sonar.java.model.VisitorsBridge;
 import org.sonar.plugins.java.api.JavaCheck;
 import org.sonar.plugins.java.api.JavaResourceLocator;
 import org.sonar.plugins.java.api.JavaVersion;
+import org.sonarsource.analyzer.commons.collections.ListUtils;
 import org.sonarsource.performance.measure.PerformanceMeasure;
 import org.sonarsource.performance.measure.PerformanceMeasure.Duration;
 
 public class JavaFrontend {
+  public static final long MAX_BATCH_SIZE = (long) (Runtime.getRuntime().totalMemory() * 0.1);
```

Comment:
With AutoScan mode, there's no semantic from project `.class` files. We should never limit the size of the batch in this mode. For now, I propose to keep the `batch size`/`memory size` computation for later.